I get those errors:```
semgrep/semgrep/target_manager.py:322: error: Argument 1 to "frozenset" has incompatible type "List[Union[str, _PathLike[str]]]"; expected "Iterable[Path]"  [arg-type]
semgrep/semgrep/target_manager.py:339: error: Argument 1 to "frozenset" has incompatible type "List[Union[str, _PathLike[str]]]"; expected "Iterable[Path]"  [arg-type]
Found 2 errors in 1 file (checked 1 source file)
```
with mypy, anyone know how to fix it? I didn't even change target_manager.py

--- a/interfaces/Input_to_core.atd
+++ b/interfaces/Input_to_core.atd
@@ -1,5 +1,5 @@
 (*
-   Type definitions for semgrep-core's JSON input, meant for the
+   Type definitions for semgrep-core's JSON input, coming from the
    python wrapper, 'semgrep'.
 
    There are other very important form of inputs which are not specified here:

--- a/interfaces/Input_to_core.atd
+++ b/interfaces/Input_to_core.atd
@@ -13,7 +13,10 @@
 (* Targets *)
 (*****************************************************************************)
 
-(* for -targets *)
+(* For -targets.
+ * coupling: if you change the type here, you probably also want to change
+ * semgrep-core/tests/e2e/target
+*)
 
 type target = {
   path: string (* source file *);

--- a/semgrep-core/src/runner/Run_semgrep.ml
+++ b/semgrep-core/src/runner/Run_semgrep.ml
@@ -429,12 +429,15 @@ let file_and_more_of_file config langs file =
 
 let targets_of_config (config : Runner_config.t) :
     In.targets * Out.skipped_target list =
+  (* TODO: add config.lang there, and check for bad combinations? *)
   match (config.target_file, config.roots) with
   | "", roots ->
       (* We usually let semgrep-python computes the list of targets (and pass it
        * via -target), but it's convenient to also run semgrep-core without
        * semgrep-python and to recursively get a list of targets.
        *)
+      (* less: could also apply Common.fullpath? *)
+      let roots = roots |> Common.map replace_named_pipe_by_regular_file in
       let lang_opt = lang_opt_of_config config in
       let files, skipped = Find_target.files_of_dirs_or_files lang_opt roots in
       let targets = files |> List.map (fun file -> { In.path = file }) in
@@ -619,7 +622,6 @@ let pattern_of_config lang config =
 *)
 let semgrep_with_one_pattern config =
   assert (config.config_file = "");
-  assert (config.target_file = "");
 
   (* TODO: support generic and regex patterns as well? See code in Deep. *)
   let lang = Xlang.lang_of_opt_xlang config.lang in
@@ -641,14 +643,9 @@ let semgrep_with_one_pattern config =
             [ minirule_of_pattern lang pattern_string pattern ])
       in
       (* simpler code path than in semgrep_with_rules *)
-      let roots =
-        (* less: could also apply Common.fullpath? *)
-        config.roots |> Common.map replace_named_pipe_by_regular_file
-      in
-      let targets, _skipped =
-        Find_target.files_of_dirs_or_files (Some lang) roots
-      in
-      targets
+      let targets, _skipped = targets_of_config config in
+      let files = targets |> List.map (fun t -> t.In.path) in
+      files
       |> List.iter (fun file ->
              logger#info "processing: %s" file;
              let process file =
@@ -677,7 +674,7 @@ let semgrep_with_one_pattern config =
       let n = List.length !E.g_errors in
       if n > 0 then pr2 (spf "error count: %d" n);
       (* TODO: what's that? *)
-      Experiments.gen_layer_maybe _matching_tokens pattern_string targets
+      Experiments.gen_layer_maybe _matching_tokens pattern_string files
 
 (*****************************************************************************)
 (* Semgrep dispatch *)

--- a/semgrep-core/tests/e2e/target
+++ b/semgrep-core/tests/e2e/target
@@ -1,1 +1,1 @@
-tests/e2e/targets/basic.py
+[ {"path": "tests/e2e/targets/basic.py"} ]

--- a/semgrep-core/tests/e2e/test_target_file.py
+++ b/semgrep-core/tests/e2e/test_target_file.py
@@ -7,7 +7,7 @@ def test_target_file():
         as having that file in -target-file flag
     """
     output = subprocess.check_output(["semgrep-core", "-e", "$X==$X", "-lang", "python", "tests/e2e/targets/basic.py"], encoding="utf-8")
-    output2 = subprocess.check_output(["semgrep-core", "-e", "$X==$X", "-lang", "python", "-target_file", "tests/e2e/target"], encoding="utf-8")
+    output2 = subprocess.check_output(["semgrep-core", "-e", "$X==$X", "-lang", "python", "-targets", "tests/e2e/target"], encoding="utf-8")
     assert output == output2
 
 

